<ui version="4.0">
 <class>DlgFilterEditClass</class>
 <widget class="QDialog" name="DlgFilterEditClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>662</width>
    <height>538</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Edit Filters</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_5">
   <item>
    <layout class="QHBoxLayout" name="layMain1">
     <item>
      <widget class="QGroupBox" name="grpFilters">
       <property name="minimumSize">
        <size>
         <width>230</width>
         <height>260</height>
        </size>
       </property>
       <property name="title">
        <string>Available Filters</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout">
        <property name="spacing">
         <number>5</number>
        </property>
        <property name="margin">
         <number>8</number>
        </property>
        <item>
         <widget class="QListView" name="lstFilters" />
        </item>
        <item>
         <layout class="QHBoxLayout" name="layFiltersActions">
          <item>
           <spacer name="spFiltersActions">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <layout class="QHBoxLayout" name="layFiltersButtons">
            <property name="spacing">
             <number>1</number>
            </property>
            <item>
             <widget class="QCheckBox" name="cbIsGlobal">
              <property name="toolTip">
               <string>Mark the filter as global or local to the project.</string>
              </property>
              <property name="text">
               <string>Global</string>
              </property>
              <property name="checked">
               <bool>false</bool> // It depends!
              </property>
             </widget>
            </item>
            <item>
             <widget class="QToolButton" name="btFiltersAdd">
              <property name="sizePolicy">
               <sizepolicy vsizetype="Fixed" hsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimumSize">
               <size>
                <width>24</width>
                <height>24</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>24</width>
                <height>24</height>
               </size>
              </property>
              <property name="text">
               <string>+</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QToolButton" name="btFiltersRem">
              <property name="sizePolicy">
               <sizepolicy vsizetype="Fixed" hsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimumSize">
               <size>
                <width>24</width>
                <height>24</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>24</width>
                <height>24</height>
               </size>
              </property>
              <property name="text">
               <string>-</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="grpColoring">
       <property name="minimumSize">
        <size>
         <width>400</width>
         <height>260</height>
        </size>
       </property>
       <property name="title">
        <string>Coloring Scheme</string>
       </property>
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <property name="spacing">
         <number>8</number>
        </property>
        <property name="margin">
         <number>8</number>
        </property>
        <item>
         <layout class="QVBoxLayout" name="layColorList">
          <item>
           <widget class="QListView" name="lstColors">
            <property name="minimumSize">
             <size>
              <width>150</width>
              <height>150</height>
             </size>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="layColorsActions">
            <item>
             <spacer name="spColorsActions">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="QToolButton" name="btColorUp">
              <property name="minimumSize">
               <size>
                <width>24</width>
                <height>24</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>24</width>
                <height>24</height>
               </size>
              </property>
              <property name="text">
               <string>△</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QToolButton" name="btColorDown">
              <property name="minimumSize">
               <size>
                <width>24</width>
                <height>24</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>24</width>
                <height>24</height>
               </size>
              </property>
              <property name="text">
               <string>▽</string>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="layColorsButtons">
              <property name="spacing">
               <number>1</number>
              </property>
              <item>
               <widget class="QToolButton" name="btColorAdd">
                <property name="sizePolicy">
                 <sizepolicy vsizetype="Fixed" hsizetype="Fixed">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="minimumSize">
                 <size>
                  <width>24</width>
                  <height>24</height>
                 </size>
                </property>
                <property name="maximumSize">
                 <size>
                  <width>24</width>
                  <height>24</height>
                 </size>
                </property>
                <property name="text">
                 <string>+</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QToolButton" name="btColorRem">
                <property name="sizePolicy">
                 <sizepolicy vsizetype="Fixed" hsizetype="Fixed">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="minimumSize">
                 <size>
                  <width>24</width>
                  <height>24</height>
                 </size>
                </property>
                <property name="maximumSize">
                 <size>
                  <width>24</width>
                  <height>24</height>
                 </size>
                </property>
                <property name="text">
                 <string>-</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QVBoxLayout" name="layColorAttributes">
          <item>
           <widget class="QGroupBox" name="grpColorFill">
            <property name="minimumSize">
             <size>
              <width>212</width>
              <height>0</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>212</width>
              <height>500</height>
             </size>
            </property>
            <property name="title">
             <string>Fill Color</string>
            </property>
            <layout class="QHBoxLayout" name="horizontalLayout_2">
             <property name="spacing">
              <number>5</number>
             </property>
             <property name="leftMargin">
              <number>5</number>
             </property>
             <property name="topMargin">
              <number>5</number>
             </property>
             <property name="bottomMargin">
              <number>5</number>
             </property>
             <item>
              <widget class="ColorEdit" native="1" name="ceFill">
               <property name="sizePolicy">
                <sizepolicy vsizetype="Fixed" hsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="minimumSize">
                <size>
                 <width>196</width>
                 <height>26</height>
                </size>
               </property>
               <property name="maximumSize">
                <size>
                 <width>196</width>
                 <height>26</height>
                </size>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QGroupBox" name="grpColorBorder">
            <property name="minimumSize">
             <size>
              <width>212</width>
              <height>0</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>212</width>
              <height>500</height>
             </size>
            </property>
            <property name="title">
             <string>Border Color</string>
            </property>
            <layout class="QHBoxLayout" name="horizontalLayout_3">
             <property name="spacing">
              <number>5</number>
             </property>
             <property name="margin">
              <number>5</number>
             </property>
             <item>
              <widget class="ColorEdit" native="1" name="ceBorder">
               <property name="sizePolicy">
                <sizepolicy vsizetype="Fixed" hsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="minimumSize">
                <size>
                 <width>196</width>
                 <height>26</height>
                </size>
               </property>
               <property name="maximumSize">
                <size>
                 <width>196</width>
                 <height>26</height>
                </size>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="layBorderWidth">
            <item>
             <spacer name="spColorBorderWidth">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>140</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="QLabel" name="lbBorderWidth">
              <property name="text">
               <string>Border Width</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="txtBorderWidth">
              <property name="maximumSize">
               <size>
                <width>50</width>
                <height>16777215</height>
               </size>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <spacer name="spColorAttributes">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="layMain2">
     <item>
      <widget class="QGroupBox" name="grpFilterSettings">
       <property name="minimumSize">
        <size>
         <width>440</width>
         <height>250</height>
        </size>
       </property>
       <property name="title">
        <string>Filter Settings</string>
       </property>
       <layout class="QGridLayout" name="gridLayout">
        <property name="margin">
         <number>8</number>
        </property>
        <property name="spacing">
         <number>5</number>
        </property>
        <item row="0" column="0">
         <widget class="QGroupBox" name="grpFilterName">
          <property name="title">
           <string>Filter Name</string>
          </property>
          <layout class="QHBoxLayout" name="layFilterName">
           <property name="leftMargin">
            <number>5</number>
           </property>
           <property name="topMargin">
            <number>4</number>
           </property>
           <property name="rightMargin">
            <number>5</number>
           </property>
           <property name="bottomMargin">
            <number>5</number>
           </property>
           <item>
            <widget class="QLineEdit" name="txtFilterName" />
           </item>
          </layout>
         </widget>
        </item>
        <item rowspan="2" row="0" column="1">
         <layout class="QVBoxLayout" name="layFilterSettingButtons">
          <item>
           <spacer name="spFilterSetButtons">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QGroupBox" name="grpFilterRuleButtons">
            <property name="title">
             <string>Filter Rule Primitives</string>
            </property>
            <layout class="QHBoxLayout" name="horizontalLayout_5">
             <property name="spacing">
              <number>5</number>
             </property>
             <property name="margin">
              <number>5</number>
             </property>
             <item>
              <layout class="QHBoxLayout" name="layFilterRuleButtons1">
               <property name="spacing">
                <number>2</number>
               </property>
               <item>
                <layout class="QVBoxLayout" name="layLogOps">
                 <property name="spacing">
                  <number>1</number>
                 </property>
                 <property name="sizeConstraint">
                  <enum>QLayout::SetFixedSize</enum>
                 </property>
                 <item>
                  <widget class="QPushButton" name="btAnd">
                   <property name="sizePolicy">
                    <sizepolicy vsizetype="Fixed" hsizetype="Fixed">
                     <horstretch>0</horstretch>
                     <verstretch>0</verstretch>
                    </sizepolicy>
                   </property>
                   <property name="maximumSize">
                    <size>
                     <width>50</width>
                     <height>22</height>
                    </size>
                   </property>
                   <property name="text">
                    <string>&lt;and&gt;</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QPushButton" name="btOr">
                   <property name="sizePolicy">
                    <sizepolicy vsizetype="Fixed" hsizetype="Fixed">
                     <horstretch>0</horstretch>
                     <verstretch>0</verstretch>
                    </sizepolicy>
                   </property>
                   <property name="maximumSize">
                    <size>
                     <width>50</width>
                     <height>22</height>
                    </size>
                   </property>
                   <property name="text">
                    <string>&lt;or&gt;</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QPushButton" name="btNot">
                   <property name="sizePolicy">
                    <sizepolicy vsizetype="Fixed" hsizetype="Fixed">
                     <horstretch>0</horstretch>
                     <verstretch>0</verstretch>
                    </sizepolicy>
                   </property>
                   <property name="maximumSize">
                    <size>
                     <width>50</width>
                     <height>22</height>
                    </size>
                   </property>
                   <property name="text">
                    <string>&lt;not&gt;</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QPushButton" name="btXor">
                   <property name="sizePolicy">
                    <sizepolicy vsizetype="Fixed" hsizetype="Fixed">
                     <horstretch>0</horstretch>
                     <verstretch>0</verstretch>
                    </sizepolicy>
                   </property>
                   <property name="maximumSize">
                    <size>
                     <width>50</width>
                     <height>22</height>
                    </size>
                   </property>
                   <property name="text">
                    <string>&lt;xor&gt;</string>
                   </property>
                  </widget>
                 </item>
                </layout>
               </item>
               <item>
                <layout class="QVBoxLayout" name="layAtoms">
                 <property name="spacing">
                  <number>1</number>
                 </property>
                 <property name="sizeConstraint">
                  <enum>QLayout::SetMinAndMaxSize</enum>
                 </property>
                 <item>
                  <widget class="QPushButton" name="btClass">
                   <property name="maximumSize">
                    <size>
                     <width>75</width>
                     <height>22</height>
                    </size>
                   </property>
                   <property name="text">
                    <string>&lt;hasClass&gt;</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QPushButton" name="btAttr">
                   <property name="maximumSize">
                    <size>
                     <width>75</width>
                     <height>22</height>
                    </size>
                   </property>
                   <property name="text">
                    <string>&lt;hasAttr&gt;</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QPushButton" name="btScore">
                   <property name="maximumSize">
                    <size>
                     <width>75</width>
                     <height>22</height>
                    </size>
                   </property>
                   <property name="text">
                    <string>&lt;score&gt;</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QPushButton" name="btSwitch">
                   <property name="maximumSize">
                    <size>
                     <width>75</width>
                     <height>22</height>
                    </size>
                   </property>
                   <property name="text">
                    <string>&lt;switch&gt;</string>
                   </property>
                  </widget>
                 </item>
                </layout>
               </item>
              </layout>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="btCompile">
            <property name="text">
             <string>Compile</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item row="1" column="0">
         <widget class="QGroupBox" name="grpFilterRule">
          <property name="title">
           <string>Filter Rule</string>
          </property>
          <layout class="QVBoxLayout" name="layFilterRule">
           <property name="leftMargin">
            <number>5</number>
           </property>
           <property name="topMargin">
            <number>4</number>
           </property>
           <property name="rightMargin">
            <number>5</number>
           </property>
           <property name="bottomMargin">
            <number>5</number>
           </property>
           <item>
            <widget class="QPlainTextEdit" name="txtFilterRule" />
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <layout class="QVBoxLayout" name="layClose1">
       <item>
        <spacer name="spClose1">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <layout class="QHBoxLayout" name="layClose2">
         <item>
          <spacer name="spClose2">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QPushButton" name="btExit">
           <property name="maximumSize">
            <size>
             <width>80</width>
             <height>24</height>
            </size>
           </property>
           <property name="text">
            <string>Close</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </item>
    </layout>
   </item>
  </layout>
  <action name="actionFilterAdd">
   <property name="text">
    <string>+</string>
   </property>
  </action>
  <action name="actionFilterRem">
   <property name="text">
    <string>-</string>
   </property>
  </action>
  <action name="actionCompile">
   <property name="text">
    <string>Compile</string>
   </property>
  </action>
  <action name="actionColorAdd">
   <property name="text">
    <string>+</string>
   </property>
  </action>
  <action name="actionColorRem">
   <property name="text">
    <string>-</string>
   </property>
  </action>
  <action name="actionColorUp">
   <property name="text">
    <string>△</string>
   </property>
  </action>
  <action name="actionColorDown">
   <property name="text">
    <string>▽</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11" />
 <customwidgets>
  <customwidget>
   <class>ColorEdit</class>
   <extends>QWidget</extends>
   <header>ColorEdit.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources />
 <connections>
  <connection>
   <sender>btExit</sender>
   <signal>clicked()</signal>
   <receiver>DlgFilterEditClass</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>531</x>
     <y>423</y>
    </hint>
    <hint type="destinationlabel">
     <x>568</x>
     <y>414</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
